junit5/src/test/java/com/hivemq/testcontainer/junit5/ContainerWithControlCenter.java

```diff
@@ -15,30 +15,34 @@
  */
 package com.hivemq.testcontainer.junit5;
 
-import com.hivemq.extension.sdk.api.annotations.NotNull;
 import org.apache.http.client.methods.HttpGet;
 import org.apache.http.client.methods.HttpUriRequest;
 import org.apache.http.impl.client.CloseableHttpClient;
 import org.apache.http.impl.client.HttpClientBuilder;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.Timeout;
-import org.junit.jupiter.api.extension.RegisterExtension;
 
 import java.util.concurrent.TimeUnit;
 
+@SuppressWarnings("ConstantConditions")
 public class ContainerWithControlCenter {
 
     public static final int CONTROL_CENTER_PORT = 8080;
 
-    @RegisterExtension
-    public @NotNull HiveMQTestContainerExtension extension =
-            new HiveMQTestContainerExtension("hivemq/hivemq4", "latest")
-            .withControlCenter(CONTROL_CENTER_PORT);
     @Test
     @Timeout(value = 5, unit = TimeUnit.MINUTES)
-    public void testHttp() throws Exception {
+    public void test() throws Exception {
+
+        final HiveMQTestContainerExtension extension =
+                new HiveMQTestContainerExtension("hivemq/hivemq4", "latest")
+                        .withControlCenter(CONTROL_CENTER_PORT);
+
+        extension.beforeEach(null);
```

Comment:
Is it right to pass null here?

Reply:
yes, the method doesn't use the argument anyway.